tests/test_settings.py

```diff
@@ -211,10 +211,44 @@ def test_font_family_chooser():
 def toplevel():
     toplevel = tkinter.Toplevel()
     toplevel.geometry("600x100")
+    toplevel.update()
     yield toplevel
     toplevel.destroy()
 
 
+def test_change_events(toplevel):
+    settings_obj = settings.Settings(toplevel, "<<ItChanged:{}>>")
+    settings_obj.add_option("foo", default="foo default")
+    settings_obj.add_option("bar", default="bar default")
+
+    change_events = []
+    toplevel.bind(
+        "<<ItChanged:foo>>",
+        (lambda e: change_events.append("foo = " + settings_obj.get('foo', str))),
+    )
+    toplevel.bind(
+        "<<ItChanged:bar>>",
+        (lambda e: change_events.append("bar = " + settings_obj.get('bar', str))),
+    )
+
+    settings_obj.set("foo", "x")
+    assert change_events == ["foo = x"]
+    settings_obj.set("bar", "y")
+    assert change_events == ["foo = x", "bar = y"]
+
+    change_events.clear()
+    with settings_obj.set_many_at_once():
+        settings_obj.set("foo", "xxx")
```

Comment:
Presumably we should only generate one `foo` change event in this case:
```suggestion
        settings_obj.set("foo", "some temporary value")
        settings_obj.set("foo", "xxx")
```